static/src/js/zustand/slices/createPreferencesSlice.ts

```diff
@@ -0,0 +1,223 @@
+      })
+    },
+
+    setPreferencesFromJwt: (jwtToken) => {
+      if (!jwtToken) return
+
+      const decoded = jwt.decode(jwtToken) as JwtTokenPayload
+      const { preferences = {} } = decoded
+
+      // TODO Remove in EDSC-4443 - Legacy layer migration
+      if (preferences.mapView) {
+        const { baseLayer, overlayLayers } = preferences.mapView
+        if (baseLayer === 'blueMarble') {
+          preferences.mapView.baseLayer = mapLayers.worldImagery
+        }
+
+        const referenceFeatureIndex = overlayLayers.indexOf('referenceFeatures')
+        const referenceLabelsIndex = overlayLayers.indexOf('referenceLabels')
+        if (referenceFeatureIndex !== -1) {
+          overlayLayers[referenceFeatureIndex] = mapLayers.bordersRoads
+        }
+
+        if (referenceLabelsIndex !== -1) {
+          overlayLayers[referenceLabelsIndex] = mapLayers.placeLabels
+        }
+      }
+
+      set((state) => {
+        state.preferences = {
+          ...state.preferences,
+          ...preferences
+        }
+      })
+
+      const { mapView: preferencesMapView = {} } = preferences
+
+      if (!isEmpty(preferencesMapView)) {
+        const currentState = get()
+        const { map: currentMap } = currentState
+        const { mapView, setMapView } = currentMap
+
+        const initialMapView = {
+          base: {
+            worldImagery: true,
+            trueColor: false,
+            landWaterMap: false
+          },
+          latitude: 0,
+          longitude: 0,
+          overlays: {
+            bordersRoads: true,
+            coastlines: false,
+            placeLabels: true
+          },
+          projection: projectionCodes.geographic,
+          rotation: 0,
+          zoom: projectionConfigs[projectionCodes.geographic].zoom
+        }
+
+        if (isEqual(mapView, initialMapView)) {
+          const {
+            baseLayer,
+            latitude,
+            longitude,
+            overlayLayers,
+            projection,
+            zoom
+          } = preferencesMapView as PreferencesState['mapView']
+
+          const base = {
+            worldImagery: false,
+            trueColor: false,
+            landWaterMap: false,
+            [baseLayer]: true
+          }
+
+          const overlays = {
+            bordersRoads: false,
+            coastlines: false,
+            placeLabels: false
+          }
+
+          overlayLayers.forEach((layer: string) => {
+            if (Object.prototype.hasOwnProperty.call(overlays, layer)) {
+              (overlays as Record<string, boolean>)[layer] = true
+            }
+          })
+
+          setMapView({
+            base,
+            latitude,
+            longitude,
+            overlays,
+            projection: projection as ProjectionCode,
+            zoom
+          })
+        }
+      }
+    },
+
+    submitAndUpdatePreferences: async (data) => {
+      const { formData: preferences } = data
+
+      get().preferences.setIsSubmitting(true)
```

Comment:
I'm wondering if we need the setIsSubmitting function, or if we just call set() here.

```
set((state) => state.preferences.isSubmitting = true)
```
I think this would do the same thing.

Should we reserve using `set*` functions for access outside of zustand, like if a component needs to set some data? Or if the setter is more complex?